server/src/infra/access_control.rs

```diff
@@ -183,41 +191,53 @@ impl<Handler: BackendHandler> AccessControlledBackendHandler<Handler> {
     pub fn get_admin_handler(
         &self,
         validation_result: &ValidationResults,
-    ) -> Option<&impl AdminBackendHandler> {
-        validation_result.is_admin().then_some(&self.handler)
```

Comment:
Specifically, this line used to be a zero-cost forwarding of a reference, and now it comes with a clone.